Update install scripts to support single RID parameters

cobya opened the pull request:
To reduce the overhead of adopting new installation script versions, allow users to only configure the runtime version without additional parameters being required to default to .NET 8. Additionally, add a warning to users for .NET Framework 3.1 installation and when lower priority parameters are selected.

![image](https://github.com/user-attachments/assets/c45cf6a0-c0c0-437a-b341-097c86e1839c)
![image](https://github.com/user-attachments/assets/ed596bdd-7fce-4fd6-9654-7cb885a8859b)

cobya commented:
/azp run

cobya commented:
/azp run

cobya commented:
/azp run

cobya commented:
/azp run

phil-hodgson commented:
The PowerShell script seems like it'd benefit from having `[CmdletBinding()]` at the beginning. Then at least we get command line help (-?) support

cobya commented:
/azp run

cobya commented:
![PowerShell Script Help](https://github.com/user-attachments/assets/7c4bfea4-30b9-456c-8d47-86ff9a7ebcb8)
Updated the script to add a warning for the upcoming 3.1 deprecation and added PS help documentation.

cobya commented:
> The PowerShell script seems like it'd benefit from having `[CmdletBinding()]` at the beginning. Then at least we get command line help (-?) support

Definitely a good idea, added it since we're doing improvement work here already.

embetten commented on helpers/installcredprovider.ps1:
```diff
@@ -122,6 +134,7 @@ if ($InstallNet8 -eq $True) {
     $zipFile = "Microsoft.Net8.${releaseRidPart}NuGet.CredentialProvider.zip"
 }
 if ($AddNetfx -eq $True) {
+    Write-Warning "The .Net Framework 3.1 version of the Credential Provider is deprecated and will be removed in the next major release. Please migrate to the .Net Framework 4.8 or .Net Core versions."
```
moving to future pr

embetten commented on helpers/installcredprovider.ps1:
```diff
@@ -42,6 +42,16 @@ if ($AddNetfx -eq $True -and $AddNetfx48 -eq $True) {
+
+    Write-Host "RuntimeIdentifier parameter is specified, the .Net 8 self-contained version will be installed"
+    $InstallNet6 = $False
+    $InstallNet8 = $True
```
yes

embetten commented on helpers/installcredprovider.ps1:
```diff
@@ -42,6 +42,16 @@ if ($AddNetfx -eq $True -and $AddNetfx48 -eq $True) {
 }
+if (![string]::IsNullOrEmpty($RuntimeIdentifier)) {
+    if (($Version.StartsWith("0.") -or $Version.StartsWith("1.0") -or $Version.StartsWith("1.1") -or $Version.StartsWith("1.2") -or $Version.StartsWith("1.3"))) {
+        Write-Error "You cannot install the .Net 8 self-contained version or with versions lower than 1.4.0"
```
Moving to future PR to unblock release
